### PR TITLE
Implement internal/net net.go&dialer.go test

### PR DESCRIPTION
**Pull request by kevindiu**
### Description:

This PR implements internal/net net.go and dialer.go test.

### Environment:

- Go Version: 1.17.3
- Docker Version: 20.10.8
- Kubernetes Version: 1.22.0
- NGT Version: 1.13.8

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/net/dialer_test.go [hunk 1]
```diff
@@ -1759,69 +1759,162 @@ func Test_dialer_tlsHandshake(t *testing.T) {
...
+					der:             &net.Dialer{},
+				},
+				checkFunc: func(w want, c *tls.Conn, e error) error {
+					if e == nil {
+						return errors.New("Handshake not completed")
+					}
+					return nil
+				},
+				afterFunc: func(a args) {
+					// srv.Close()
```
Comment (vankichi) [inline internal/net/dialer_test.go:1912]:
```suggestion
```

Comment (vankichi) [inline internal/net/dialer_test.go:1912]:
```suggestion
```

File: internal/net/dialer_test.go [hunk 2]
```diff
@@ -1759,69 +1759,162 @@ func Test_dialer_tlsHandshake(t *testing.T) {
...
+					conn.Close()
+					cancel()
+				},
+			}
+		}(),
```
Comment (vankichi) [inline internal/net/dialer_test.go:1917]: could you implement without an initial struct for test fields?

Comment (kevindiu) [reply internal/net/dialer_test.go]: fixed. can you please review it? thank you!

**Review by vankichi — DISMISSED** [review]

**Review by kmrmt — DISMISSED** [review]

**Pull request merged**